lib/_tls_wrap.js

```diff
@@ -989,7 +989,10 @@ exports.connect = function(/* [port, host], options, cb */) {
     var verifyError = socket._handle.verifyError();
 
     // Verify that server's identity matches it's certificate's names
-    if (!verifyError) {
+    // Unless server has resumed our existing session
+    if (!verifyError &&
+        (!options.session ||
+         Buffer.compare(options.session, socket.getSession()) !== 0)) {
```

Comment:
FWIW `Buffer.compare()` is still fairly slow right now and using a manual for-loop in js land is faster.